codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttMessageBuilders.java

```diff
@@ -546,25 +546,25 @@ public PubAckBuilder properties(MqttProperties properties) {
             return this;
         }
 
-        public MqttMessage build() {
+        public MqttPubAckMessage build() {
             MqttFixedHeader mqttFixedHeader =
                     new MqttFixedHeader(MqttMessageType.PUBACK, false, MqttQoS.AT_MOST_ONCE, false, 0);
             MqttPubReplyMessageVariableHeader mqttPubAckVariableHeader =
                     new MqttPubReplyMessageVariableHeader(packetId, reasonCode, properties);
-            return new MqttMessage(mqttFixedHeader, mqttPubAckVariableHeader);
+            return new MqttPubAckMessage(mqttFixedHeader, mqttPubAckVariableHeader);
         }
     }
 
     public static final class SubAckBuilder {
 
-        private short packetId;
+        private int packetId;
         private MqttProperties properties;
         private final List<MqttQoS> grantedQoses = new ArrayList<MqttQoS>();
 
         SubAckBuilder() {
         }
 
-        public SubAckBuilder packetId(short packetId) {
+        public SubAckBuilder packetId(int packetId) {
```

Comment:
I think we need to keep the old method as well as otherwise this would be a breaking change